client/deployment.go

```diff
@@ -116,26 +122,67 @@ func (d *Deployment) AddUser(username, password string, admin bool) (*http.Respo
 		Password: password,
 		Admin:    admin,
 	}
-	return d.request("POST", "/user/adduser", reqContent)
+	return d.post("/user/adduser", reqContent)
 }
 
 // RemoveUser prevents a user from accessing Inertia Web
 func (d *Deployment) RemoveUser(username string) (*http.Response, error) {
 	reqContent := &common.UserRequest{Username: username}
-	return d.request("POST", "/user/removeuser", reqContent)
+	return d.post("/user/removeuser", reqContent)
 }
 
 // ResetUsers resets all users on the remote.
 func (d *Deployment) ResetUsers() (*http.Response, error) {
-	return d.request("POST", "/user/resetusers", nil)
+	return d.post("/user/resetusers", nil)
 }
 
 // ListUsers lists all users on the remote.
 func (d *Deployment) ListUsers() (*http.Response, error) {
-	return d.request("GET", "/user/listusers", nil)
+	return d.get("/user/listusers", nil)
+}
+
+// Sends a GET request. "queries" contains query string arguments.
+func (d *Deployment) get(endpoint string, queries map[string]string) (*http.Response, error) {
+	// Assemble URL
+	url, err := url.Parse("https://" + d.RemoteVPS.GetIPAndPort())
+	if err != nil {
+		return nil, err
+	}
+	url.Path = path.Join(url.Path, endpoint)
+	urlString := url.String()
+
+	// Assemble request
+	req, err := http.NewRequest("GET", urlString, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add query strings
+	if queries != nil {
+		q := req.URL.Query()
+		for k, v := range queries {
+			q.Add(k, v)
+		}
+		req.URL.RawQuery = q.Encode()
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+d.Auth)
+
+	// Make HTTPS request
+	tr := &http.Transport{
+		// Our certificates are self-signed, so will raise
+		// a warning - currently, we ask our client to ignore
+		// this warning.
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+	client := &http.Client{Transport: tr}
```

Comment:
The HTTP clients (and transports) are the same for all request types - could we have it set up and fetched from a helper function?